### PR TITLE
add @nuxt/image

### LINKED ISSUE #7807: add @nuxt/image
Try tune and leverage all performance things we could use

- https://image.nuxt.com/
- https://image.nuxt.com/get-started/installation
- https://image.nuxt.com/providers/cloudflare

Ref
---
- https://github.com/kodadot/nft-gallery/issues/4716
- https://github.com/kodadot/nft-gallery/issues/4551

### PR DESCRIPTION
**Pull request by pbkompasz**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [X] Feature
- [ ] Refactoring

## Context

- [X] Closes #7807
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [X] My contribution builds **clean without any errors or warnings**
- [X] I've merged recent default branch -- **main** and I've no conflicts
- [X] I've tried to respect high code quality standards
- [X] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [X] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16hqk8Tjugo9AwSzKkxXc1dsRK7iJetpx1YBaoXYwdHzuNAt&usdamount=50&donation=true)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### <samp>🤖 Generated by Copilot at 4ebdf8f</samp>

This pull request adds the `@nuxt/image` module to the project and updates the `pnpm-lock.yaml` and `package.json` files accordingly. The `@nuxt/image` module enables image optimization and resizing for the NFT gallery website, which improves the performance and quality of the images.

### <samp>🤖 Generated by Copilot at 4ebdf8f</samp>

> _To make images look more sublime_
> _We added the `@nuxt/image` module this time_
> _It optimizes and resizes_
> _And the lock file revises_
> _The dependencies that need to build fine_

### CODE REVIEW

**Comment by yangwao** [comment]
woah testing it and I've noticed speed improvements, don't see empty windows when scrolling fast down on Kusama Asset Hub, thank you, keep up great work please!

**Comment by yangwao** [comment]
Only bug I've found generative collections doesn't work through it 

image

https://deploy-preview-7924--koda-canary.netlify.app/ahk/gallery/176-25

**Comment by yangwao** [comment]
I noticed Lighthouse score is quite great here
image

upcoming scores will be these if merged
- https://github.com/kodadot/nft-gallery/pull/8030#issuecomment-1808371703

**Comment by pbkompasz** [comment]
> I noticed Lighthouse score is quite great here1808371703)

@yangwao Yes it did.
I will try to finish it by tomorrow.

**Comment by yangwao** [comment]
I've found some pictures doesn't load on landing, maybe merge main if there are some changes? 

image

**Comment by pbkompasz** [comment]
@yangwao I've seen that too, I got a 404 for some of the images.
When the image is not available should I put the placeholder and above it the name of the image?

**Comment by yangwao** [comment]
> When the image is not available should I put the placeholder and above it the name of the image?

I think most of images should be available as on canary.kodadot.xyz it works 👀 otherwise yes

File: libs/ui/src/components/MediaItem/type/ImageMedia.vue [hunk 1]
```diff
@@ -5,7 +5,14 @@
       'is-square image': !original,
       'is-detail': isDetail,
     }">
-    <img
+    <NuxtImg
```
Comment (roiLeo) [inline libs/ui/src/components/MediaItem/type/ImageMedia.vue:8]: shouldn't be available here

File: nuxt.config.ts [hunk 2]
```diff
@@ -11,30 +10,13 @@ export default defineNuxtConfig({
...
-  sourcemap: false,
-
   vue: {
     compilerOptions: {
       // model-viewer from ModelMedia throw warning
       isCustomElement: (tag) => tag.includes('model-viewer'),
     },
   },
 
-  vite: {
```
Comment (roiLeo) [inline nuxt.config.ts:23]: why remove?

**Comment by shashkovdanil** [comment]
@yangwao okay I'll take it

**Comment by shashkovdanil** [comment]
@yangwao current status. Images don't resize on cloudflare, get error 9524

```
9524 — The /cdn-cgi/image/ resizing service could not perform resizing. This may happen when an image URL is intercepted by a Worker. Check your Workers and try again. This can also happen when using a pages.dev URL of a Cloudflare Pages project. In that case, you can use a Custom Domain instead.
```

It seems that we have this problem
"This may happen when an image URL is intercepted by a Worker."

I don't have access to cloudflare dashboard, so someone have to fix it. Without solving this problem, adding nuxt-image is pointless, it won't do any good

cc @preschian @roiLeo maybe you can help with it

**Comment by preschian** [comment]
> @yangwao current status. Images don't resize on cloudflare, get [error 9524](https://developers.cloudflare.com/images/image-resizing/troubleshooting/#:~:text=9524%20%E2%80%94%20The%20%2Fcdn%2Dcgi,happen%20when%20using%20a%20pages.)
> 
> ```
> 9524 — The /cdn-cgi/image/ resizing service could not perform resizing. This may happen when an image URL is intercepted by a Worker. Check your Workers and try again. This can also happen when using a pages.dev URL of a Cloudflare Pages project. In that case, you can use a Custom Domain instead.
> ```
> 
> It seems that we have this problem `This may happen when an image URL is intercepted by a Worker.`
> 
> I don't have access to cloudflare dashboard, so someone have to fix it. Without solving this problem, adding nuxt-image is pointless, it won't do any good
> 
> cc @preschian @roiLeo maybe you can help with it

Send me the link, will check

The format that we can use is something like this

```
https://kodadot.xyz/cdn-cgi/imagedelivery/jk5b6spi_m_-9qC4VTnjpg/bafybeic3h44kpgwmp2l6lwzuchl7cqlscjx7opcjkdbmwpc2nzxkcublla/public

https://kodadot.xyz/cdn-cgi/imagedelivery/jk5b6spi_m_-9qC4VTnjpg/{cid-here}/public
```

Please note:
- on the preview page, we are using netlify, so make sure to give the full URL `kodadot.xyz/cdn-cgi/{image-path}` instead of `/cdn-cgi/{image-path}`
- set fallback image `onError` to current image worker

**Comment by shashkovdanil** [comment]
@preschian

Url format from cloudflare docs https://developers.cloudflare.com/images/image-resizing/url-format/

cloudflare provider uses this format https://kodadot.xyz/cdn-cgi/image/q=85,w=100,h=100/https://image.w.kodadot.xyz/ipfs/bafkreiffld3244lhtkzmkhipi6332t6yvv4nmkbfktaqerubkuht7uwgli

and this format doesn't work.

Your option `https://kodadot.xyz/cdn-cgi/imagedelivery/jk5b6spi_m_-9qC4VTnjpg/bafybeic3h44kpgwmp2l6lwzuchl7cqlscjx7opcjkdbmwpc2nzxkcublla/blur=50` is good and it works, but we can't use default cloudflare provider, because we will need to convert this `https://image.w.kodadot.xyz/ipfs/bafybeic3h44kpgwmp2l6lwzuchl7cqlscjx7opcjkdbmwpc2nzxkcublla` to this `imagedelivery/jk5b6spi_m_-9qC4VTnjpg/bafybeic3h44kpgwmp2l6lwzuchl7cqlscjx7opcjkdbmwpc2nzxkcublla`.

**Comment by preschian** [comment]
> Url format from cloudflare docs [developers.cloudflare.com/images/image-resizing/url-format](https://developers.cloudflare.com/images/image-resizing/url-format/)

Yes, this is the format that we use. from cf-docs also https://developers.cloudflare.com/images/cloudflare-images/transform/flexible-variants/

> and this format doesn't work.

That's why in this PR, @pbkompasz tried to create a custom provider

**Comment by shashkovdanil** [comment]
@preschian ah, okay, cool, I've created custom provider, will do PR soon

**Comment by preschian** [comment]
> Url format from cloudflare docs [developers.cloudflare.com/images/image-resizing/url-format](https://developers.cloudflare.com/images/image-resizing/url-format/)
> 
> cloudflare provider uses this format [kodadot.xyz/cdn-cgi/image/q=85,w=100,h=100/https://image.w.kodadot.xyz/ipfs/bafkreiffld3244lhtkzmkhipi6332t6yvv4nmkbfktaqerubkuht7uwgli](https://kodadot.xyz/cdn-cgi/image/q=85,w=100,h=100/https://image.w.kodadot.xyz/ipfs/bafkreiffld3244lhtkzmkhipi6332t6yvv4nmkbfktaqerubkuht7uwgli)

anyway, this format also supported. for ipfs, only works with https://kodadot1.infura-ipfs.io origin

- https://kodadot.xyz/cdn-cgi/image/blur=50/https://kodadot1.infura-ipfs.io/ipfs/bafybeifpzxrmi5zfvrlmoudih3l73fgcmxpybj2tmga45wogidtqhibno4/S01_Map_Coll_Logo.png
- https://kodadot.xyz/cdn-cgi/image/blur=50/https://polkadot-data.s3.us-east-2.amazonaws.com/metadata/nfts-88/nfts-88_item-5413-img.jpg

**Comment by shashkovdanil** [comment]
@yangwao @preschian PR is here https://github.com/kodadot/nft-gallery/pull/8185

I can't assign a task to me because it's on pbkompasz

**Comment by yangwao** [comment]
Meanwhile done.
Thanks, @pbkompasz for your contribution so far, yet we prefer work to be done fast 
- https://github.com/kodadot/nft-gallery/pull/8185

**Pull request closed without merging**